lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java

```diff
@@ -3,46 +3,79 @@
 import com.mapzen.android.lost.api.FusedLocationProviderApi;
 import com.mapzen.android.lost.api.LocationListener;
 import com.mapzen.android.lost.api.LocationRequest;
+import com.mapzen.android.lost.api.LostApiClient;
 
 import android.app.PendingIntent;
+import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
+import android.content.ServiceConnection;
 import android.location.Location;
+import android.os.IBinder;
 import android.os.Looper;
 import android.util.Log;
 
 import java.io.File;
-import java.util.HashMap;
 import java.util.Map;
 
 /**
  * Implementation of the {@link FusedLocationProviderApi}.
  */
 public class FusedLocationProviderApiImpl
-    implements FusedLocationProviderApi, LocationEngine.Callback {
+    implements FusedLocationProviderApi {
 
   private static final String TAG = FusedLocationProviderApiImpl.class.getSimpleName();
 
   private final Context context;
-  private boolean mockMode;
-  private LocationEngine locationEngine;
-  private Map<LocationListener, LocationRequest> listenerToRequest;
-  private Map<PendingIntent, LocationRequest> intentToRequest;
+  private FusedLocationProviderService service;
+
+  private final ServiceConnection serviceConnection = new ServiceConnection() {
+    @Override public void onServiceConnected(ComponentName name, IBinder binder) {
+      FusedLocationProviderService.FusedLocationProviderBinder fusedBinder =
+          (FusedLocationProviderService.FusedLocationProviderBinder) binder;
+      if (fusedBinder != null) {
+        service = fusedBinder.getService();
+      }
+
+      if (connectionCallbacks != null) {
+        connectionCallbacks.onConnected();
+      }
+      Log.d(TAG, "[onServiceConnected]");
+    }
+
+    @Override public void onServiceDisconnected(ComponentName name) {
+      if (connectionCallbacks != null) {
+        connectionCallbacks.onConnectionSuspended();
+      }
+      Log.d(TAG, "[onServiceDisconnected]");
+    }
+  };
+
+  LostApiClient.ConnectionCallbacks connectionCallbacks;
 
   public FusedLocationProviderApiImpl(Context context) {
     this.context = context;
-    locationEngine = new FusionEngine(context, this);
-    listenerToRequest = new HashMap<>();
-    intentToRequest = new HashMap<>();
+  }
+
+  public void connect(LostApiClient.ConnectionCallbacks callbacks) {
+    Intent intent = new Intent(context, FusedLocationProviderService.class);
+    context.startService(intent);
+
+    connectionCallbacks = callbacks;
+    intent = new Intent(context, FusedLocationProviderService.class);
+    context.bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
+  }
+
+  public void disconnect() {
+    context.unbindService(serviceConnection);
```

Comment:
...continuing from this last thought...If we keep the explicit starting of the service in place, this use case will work. So maybe we SHOULD keep the explicit starting of the service and add a call to stop the service as you suggest. I was originally thinking that this could lead to unexpected behavior if there were multiple clients because if one client called `disconnect()` it would stop the service that others were bound to but I realize now that there will only ever be one client bound to the service because of L27 in LostApiClientImpl.

So lets do as you say and add a call to explicitly stop the service... 👍